lanweatherd/main.cpp

```diff
@@ -0,0 +1,83 @@
+#include <cstdlib>
+#include <cstring>
+#include <stdlib.h>
+#include <string>
+#include <unistd.h>
+#include <sys/stat.h>
+#include <syslog.h>
+#include <thread>
+#include <vector>
+
+
+using namespace std;
+
+int *nop(void) {
+    syslog(LOG_NOTICE, "called nop");
+    return 0;
+}
+
+int main(void) {
+    // TODO: check for existing instance of the daemon
+
+    pid_t pid, sid;
+
+    pid = fork();
+
+    if (pid > 0) {
+        // it's the parent, it can exit
+        exit(EXIT_SUCCESS);
+    } else if (pid < 0) {
+        // there was an error forking, exit
+        exit(EXIT_FAILURE);
+    }
+
+    // the child process continues from here
+
+    // set default file permissions
+    umask(0);
+
+    // tap into the system log
+    openlog("lanweatherd", LOG_NOWAIT | LOG_PID, LOG_USER);
+    syslog(LOG_NOTICE, "started LAN Weather daemon");
+
+    // session id
+    sid = setsid();
+    if (sid < 0) {
+        syslog(LOG_ERR, "could not generate session id");
+        exit(EXIT_FAILURE);
+    }
+
+    // move to the filesystem root
+    if (chdir("/")) {
+        syslog(LOG_ERR, "failed to move to /");
+        exit(EXIT_FAILURE);
+    }
+
+    // close stdin, stdout, stderr
+    close(STDIN_FILENO);
+    close(STDOUT_FILENO);
+    close(STDERR_FILENO);
+
+    std::vector<std::thread> threads;
```

Comment:
Is using "std::EXAMPLE" needed here? I don't think it's required because you set it as a namespace earlier.

Reply:
(Same for all the other places it's used, forgot to add that part)

Reply:
you're probably right. i never exactly learned how to use namespaces and am basically writing c+ instead of c++ right now.